node/src/proxy_server/mod.rs

```diff
@@ -642,23 +642,31 @@ impl ProxyServer {
         }
     }
 
-    fn find_or_generate_stream_key(&mut self, ibcd: &InboundClientData) -> StreamKey {
+    fn find_or_generate_stream_key(
+        &mut self,
+        ibcd: &InboundClientData,
+        server_addr: SocketAddr,
+    ) -> StreamKey {
         match self.keys_and_addrs.b_to_a(&ibcd.peer_addr) {
             Some(stream_key) => {
                 debug!(
                     self.logger,
-                    "make_stream_key() retrieved existing key {} for {}",
+                    "find_or_generate_stream_key() retrieved existing key {} for {}",
                     &stream_key,
                     ibcd.peer_addr
                 );
                 stream_key
             }
             None => {
-                let stream_key = self.stream_key_factory.make();
+                let stream_key = self
+                    .stream_key_factory
+                    .make(self.main_cryptde.public_key(), server_addr);
```

Comment:
Hmm. Maybe the `StreamKeyFactory` should carry around the Node's public key? It's not like it's going to change...

Reply:
We already talked about this, but for posterity: no. `InboundClientData.peer_addr` (now called `InboundClientData.client_addr`) is the address associated with the stream from the browser (or other client) we're dealing with. Hashed with our public key, it ought to be universally unique.